Source/Core/Common/FileUtil.cpp

```diff
@@ -276,33 +279,68 @@ bool DeleteDir(const std::string& filename)
   return false;
 }
 
+// Repeatedly invokes func until it returns true or max_attempts failures.
+// Waits after each failure, with each delay doubling in length.
+template <typename FuncType>
+static bool AttemptMaxTimesWithExponentialDelay(const FuncType& func, int max_attempts,
+                                                std::chrono::milliseconds delay)
+{
+  for (int failed_attempts = 0; failed_attempts < max_attempts; ++failed_attempts)
+  {
+    if (func())
+    {
```

Comment:
To make this less noisy, I'd get rid of the braces here.

Reply:
I know Dolphin has a lot of them but I'm not a fan of braceless conditionals/loops; they reduce vertical spacing but allow for things like the Apple SSL bug. Unless it's a deal-breaker I'd rather leave them as is.

This would be a nice place for a Ruby-style "return if [condition]" statement, if only c++ had those.